lib/msprime.c

```diff
@@ -5823,6 +5843,18 @@ msp_add_mass_migration(
     return ret;
 }
 
+static void
+msp_deactivate_population(msp_t *self, int population_id)
```

Comment:
Just thinking out loud -- are there any 'global' checks that need to be done before you deactivate a population, perhaps to do with the populations that the extant lineages belong to? If so they'd go in here, right?

Reply:
Also, is there a reason why there isn't also an an msp_activate_population function? Because I can imagine some general assertions that might need to be checked before doing this too -- for example, if there are already samples in the simulation belonging to the population to be 'activated'

Reply:
Good ideas - I've added a bug_assert for the number of lineages in the deactivated pop == 0 and added a "activate" population. In the interest of keeping things simple I've focused on just maintaining the state machine here - we catch all the other errors if we try to migrate/sample any lineages into a populations where state != active.